keps/sig-api-machinery/20190923-extract-code-generator.md

```diff
@@ -0,0 +1,154 @@
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core Kubernetes i.e., [kubernetes/kubernetes], we require the following Release Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+k8s.io/code-generator is logically a separate component and one that we have structured as a staging repo before extraction.
+After we make the k8s.io/code-generator authoritative, we will vendor the repo into k8s.io/kubernetes to perform our generation.
+
+## Motivation
+
+The k8s.io/code-generator is externally valuable and there is no compelling reason to keep it in tree.
+We want to become a standard consumer of generator like all the others. 
+
+### Goals
+
+1. k8s.io/code-generator is the authoritative location of the kube code-generator.
+
+### Non-Goals
+
+1. Changing the k8s.io/code-generator
+
+## Proposal
+
+k8s.io/code-generator only depends on gengo, klog, and kube-openapi (from [go.sum](https://github.com/kubernetes/code-generator/blob/master/go.sum#L106-L114)).
+Because it was in staging, its history is up to date.  This means we can simply...
+
+1. Stop publishing k8s.io/code-generator
+2. Vendor k8s.io/code-generator as normal using go.mod.
```

Comment:
How would the release process/tags look for k/code-generator when it's moved out? Right now, the publishing-bot creates `kubernetes-1.x.y` (and soon `v0.x.y`) tags automatically.